blog/2024-11-26-robotics-mongodb/index.mdx

```diff
@@ -0,0 +1,108 @@
+
+## Development Experience
+
+In this section, we will compare the development experience of MongoDB and ReductStore, focusing on their query languages, communication protocols, and integration capabilities.
+
+### Query languages and SDK support
+
+[**ReductStore supports querying data in several different ways**](/docs/guides/data-querying) – either by using its _CLI, SDKs_, or _HTTP API_. This database solution improves data retrieval by batching records within a time range, minimizing requests and delays. Apart from querying by a time range, there is a possibility to retrieve a record by its timestamp, but you have to keep in mind that such querying is less efficient due to the lack of batching. ReductStore provides client SDKs for Python, JavaScript, C++, and Rust.
+
+**MongoDB** uses its own query language, _MongoDB Query Language (MQL),_ which allows for flexible and powerful data querying. MQL supports complex filtering, aggregation, and indexing, making it a great choice for applications that need to work with semi-structured data. MongoDB also provides official SDKs for various languages, such as Java, Python, and C#, making it easy to integrate with various development environments.
+
+### Communication protocols
+
+**ReductStore** uses _HTTP/HTTPS protocols_ to handle data, focusing on reducing network load in environments with high latency. It processes large datasets by batching requests and delivering data in segments, making it efficient for both real-time and historical data access.
+
+On the other hand, **MongoDB** uses its native _MongoDB Wire Protocol_, a lightweight, binary protocol optimized for fast communication between clients and servers. This protocol encodes data in BSON (Binary JSON), which supports efficient transmission and storage of JSON-like documents. These design choices make MongoDB a great choice for web-based applications, offering both flexibility and scalability.
+
+## Data Management Features
+
+### Data batching capabilities\*
+
+**ReductStore** enables efficient _data batching_ by grouping multiple records into a single HTTP request, minimizing delays and reducing the number of queries needed. It uses parameters like each_s (records per second) or each_n (specific record counts) to customize data retrieval, allowing applications to process only the required subsets efficiently. This batching mechanism supports use cases like real-time monitoring and historical data analysis with minimal network overhead.
```

Comment:
each_s and each_n is involved into querying but not batching. The storage batched data records into an HTTP request in an optimal way for the HTTP protocol